packages/ai/core/embed/embed-many.test.ts

```diff
@@ -18,6 +18,68 @@ const testValues = [
   'snowy night in the mountains',
 ];
 
+describe('model.supportsParallelCalls', () => {
+  it('should not parallelize when false', async () => {
+    let i = 0;
+
+    const { embeddings } = await embedMany({
+      model: new MockEmbeddingModelV2({
+        supportsParallelCalls: false,
+        maxEmbeddingsPerCall: 1,
+        doEmbed: async ({ values }) => {
+          const index = i++;
+
+          expect(values).toEqual([testValues[index]]);
+
+          await new Promise(resolve => setTimeout(resolve, 1));
+
+          expect(i).toEqual(index + 1);
+
+          return {
+            embeddings: [dummyEmbeddings[index]],
+            response: { headers: {}, body: {} },
+          };
+        },
+      }),
+      values: testValues,
+    });
+
+    expect(embeddings.length).toEqual(testValues.length);
+  });
+
+  it('should parallelize when true', async () => {
+    let i = 0;
+
+    const { embeddings } = await embedMany({
+      model: new MockEmbeddingModelV2({
+        supportsParallelCalls: true,
+        maxEmbeddingsPerCall: 1,
+        doEmbed: async ({ values }) => {
+          const index = i++;
+
+          expect(values).toEqual([testValues[index]]);
+
+          await new Promise(resolve => setTimeout(resolve, 1));
```

Comment:
same - can we do this without delays that tend to be brittle? e.g. by resolving mock promises? (that's how i usually deal with parallel vs non-parallel without needing the brittleness of timeouts)